dir.c

```diff
@@ -2955,42 +2955,33 @@ dir_s_glob(rb_execution_context_t *ec, VALUE obj, VALUE str, VALUE rflags, VALUE
 static VALUE
 dir_open_dir(int argc, VALUE *argv)
 {
-    VALUE dir = rb_funcallv_kw(rb_cDir, rb_intern("open"), argc, argv, RB_PASS_CALLED_KEYWORDS);
+    VALUE dir = rb_funcallv_kw(rb_cDir, rb_intern("open"), argc, argv, RB_PASS_KEYWORDS);
 
     rb_check_typeddata(dir, &dir_data_type);
     return dir;
 }
 
+static void
+dir_open_argv(VALUE *argv, VALUE path, VALUE enc)
+{
+    VALUE kw_hash = rb_hash_new();
+    rb_gc_register_mark_object(kw_hash);
+    rb_hash_aset(kw_hash, ID2SYM(rb_intern("encoding")), enc);
+    rb_obj_freeze(kw_hash);
```

Comment:
The purpose to move C-methods to builtin methods is to eliminate overheads of hash creation for keywords argument.
Making a hash here loses it.

Reply:
Remove and Fix it.